useful/replace.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+find ../ -name "parsers.yaml" -print0 | xargs -0 sed -i 's/---/---\nclass: RegexParser\nlog_type : default/g'
+find ../ -name "rules*.yaml" -print0 | xargs -0 sed -i 's/cause: \([a-z]*\)/causes: [\1]/g'
+echo -e "---\n['RegexFilenameParser', {'type': 'default', 'path_pattern': 'node_1.log'}]" > log_locations.yaml
+for i in $( find ../whylog/ -name "[0-9]*" -type d); do
```

Comment:
This syntax fails if there is any kind of whitespace in the filename. While here you protect against it by using a regex, it is generally advised to use find's `-exec` for invocation of a command for each item found by `find`.